0007-configuration-management-extension/README.md

```diff
@@ -87,34 +115,139 @@ The server MAY send configurations to the endpoint when no request were done bef
 
 If the current config matches one specified by `configId` in the request, the server MUST return response with both `configId` and `config` absent.
 
-If an endpoint successfully applies provided configuration, it SHOULD notify the server via `/applied` resource.
-
-### Applied resource
-`/<endpoint_token>/applied/json` resource is used by the endpoint to notify the server of successfully applied configuration.
-
-The request is a JSON object with the following fields:
-- `configId` (required) - id of applied configuration.
-
-The response is empty.
+Examples:
+```json
+{
+  "configId": "97016dbe8bb4adff8f754ecbf24612f2"
+}
+```
+
+```json
+{
+  "observe": true
+}
+```
+
+### Response
+Response payload MUST be a UTF-8 encoded object with the JSON Scheme as defined in [config-response.schema.json](./config-response.schema.json) file.
+
+```json
+{
+  "$schema": "http://json-schema.org/schema#",
+  "title": "7/CMX config response schema",
+
+  "type": "object",
+  "properties": {
+    "configId": {
+      "type": "string",
+      "description": "Identifier of the current configuration."
+    },
+    "config": {
+      "description": "Configuration body of an arbitrary type (as set by the request `config_format`)."
+    }
+  },
+  "additionalProperties": false
+}
+```
 
-## Open questions
-### Merge resources
-The request to `/config` resource already includes all fields for `/applied` requests, so it might be possible to merge them in a single resource.
-
-### Error handling
-Whaw errors are possible here? Should endpoint care?
-
-### Configuration rejection
-Should it be possible for the endpoint to actively reject a configuration?
+`configId` and `config` MUST come in a pair. If `configId` and `config` are absent in the response, that means configuration hasn't changed.
 
-Examples are:
-- the endpoint is unable to process the configuration.
-- the configuration is ill-formated, or pre-conditions are not met.
+If an endpoint successfully applies provided configuration, it SHOULD notify the server via `/applied` resource.
 
-### Subscribing to a part of configuration
+Examples:
+- configuration hasn't changed
+  ```json
+  {
+  }
+  ```
```

Comment:
Not sure what you propose. 1/KP only includes statusCode for errors, and 304 is clearly not an error.

The only way to put 304 in there is to add `statusCode` field. But it can only be 200 or 304 in success response, so that has little sense.

Though, there are status codes in CoAP, and we can use `2.03 Valid` code.